pinot-common/src/main/java/org/apache/pinot/common/response/CursorResponse.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.response;
+
+
+public interface CursorResponse extends BrokerResponse {
+
+  void setBrokerHost(String brokerHost);
+
+    /**
+     * get hostname of the processing broker
+     * @return String containing the hostname
+     */
+  String getBrokerHost();
+
+  void setBrokerPort(int brokerPort);
+
+    /**
+     * get port of the processing broker
+     * @return int containing the port.
+     */
+  int getBrokerPort();
+
+  /**
+   * Set the starting offset of result table slice
+   * @param offset Offset of the result table slice
+   */
+  void setOffset(int offset);
+
+  /**
+   * Set the number of rows in the result table slice.
+   * @param numRows Number of rows in the result table slice
+   */
+  void setNumRows(int numRows);
+
+  /**
+   * Current offset in the query result.
+   * Starts from 0.
+   * @return current offset.
+   */
+  int getOffset();
```

Comment:
nit: put the methods for the same object together, i.e. swapping the `setNumRows()` and `getOffset()` methods

Reply:
Done